app/controllers/homepage_controller.rb

```diff
@@ -7,28 +7,32 @@ class HomepageController < ApplicationController
   VIEW_TYPES = ["grid", "list", "map"]
 
   def index
-    redirect_to landing_page_path and return if no_current_user_in_private_clp_enabled_marketplace?
-
-    @homepage = true
+    all_shapes = shapes.get(community_id: @current_community.id)[:data]
+    shape_name_map = all_shapes.map { |s| [s[:id], s[:name]]}.to_h
 
-    @view_type = HomepageController.selected_view_type(params[:view], @current_community.default_browse_view, APP_DEFAULT_VIEW_TYPE, VIEW_TYPES)
+    if FeatureFlagHelper.search_engine == :discovery
+      @view_type = "grid"
+    else
+      @view_type = HomepageController.selected_view_type(params[:view], @current_community.default_browse_view, APP_DEFAULT_VIEW_TYPE, VIEW_TYPES)
+      @big_cover_photo = !(@current_user || CustomLandingPage::LandingPageStore.enabled?(@current_community.id)) || params[:big_cover_photo]
 
-    @big_cover_photo = !(@current_user || CustomLandingPage::LandingPageStore.enabled?(@current_community.id)) || params[:big_cover_photo]
+      @categories = @current_community.categories.includes(:children)
+      @main_categories = @categories.select { |c| c.parent_id == nil }
 
-    @categories = @current_community.categories.includes(:children)
-    @main_categories = @categories.select { |c| c.parent_id == nil }
+      # This assumes that we don't never ever have communities with only 1 main share type and
+      # only 1 sub share type, as that would make the listing type menu visible and it would look bit silly
+      listing_shape_menu_enabled = all_shapes.size > 1
+      @show_categories = @categories.size > 1
+      show_price_filter = @current_community.show_price_filter && all_shapes.any? { |s| s[:price_enabled] }
 
-    all_shapes = shapes.get(community_id: @current_community.id)[:data]
+      filters = @current_community.custom_fields.where(search_filter: true).sort
+      @show_custom_fields = filters.present? || show_price_filter
+      @category_menu_enabled = @show_categories || @show_custom_fields
+    end
 
-    # This assumes that we don't never ever have communities with only 1 main share type and
-    # only 1 sub share type, as that would make the listing type menu visible and it would look bit silly
-    listing_shape_menu_enabled = all_shapes.size > 1
-    @show_categories = @categories.size > 1
-    show_price_filter = @current_community.show_price_filter && all_shapes.any? { |s| s[:price_enabled] }
+    redirect_to landing_page_path and return if no_current_user_in_private_clp_enabled_marketplace?
```

Comment:
These kind of redirections should be the first line I think - no need to execute the function any further